File: services/gateway/internal/domain/dto/review.go
package dto

type ReviewRequestDTO struct {
	TaskID         string `json:"task_id" binding:"required"`
	RepositoryLink string `json:"repository_link" binding:"required"`
}

type ReviewResponseDTO struct {
	TaskID         string `json:"task_id"`
	Rating         string `json:"rating"`
	RepositoryLink string `json:"repository_link"`
}

File: services/telegram-bot/src/domain/dtos/review.py
from dataclasses import dataclass
from typing import Self

from src.domain.enums.review import ReviewStatusEnum


@dataclass
class ReviewRequestDTO:
    task_id: str
    repository_link: str


@dataclass
class ReviewResponseDataDTO:
    task_id: str
    overview: str
    repository_link: str


@dataclass
class ReviewResponseDTO:
    status: ReviewStatusEnum
    data: ReviewResponseDataDTO | None = None

    @classmethod
    def validate(cls, obj: dict[str, str]) -> Self:
        data = (
            ReviewResponseDataDTO(
                task_id=obj["task_id"],
                overview=obj["overview"],
                repository_link=obj["repository_link"],
            )
            if obj["status"] == ReviewStatusEnum.READY.value
            else None
        )
        return ReviewResponseDTO(
            status=obj["status"],
            data=data,
        )
